reinforcement_learning/bindings/cs/rl.net/LiveModel.cs

```diff
@@ -49,10 +53,24 @@ public bool TryChooseRank(string actionId, string contextJson, RankingResponse r
             return result == NativeMethods.SuccessStatus;
         }
 
-        public bool TryReportOutcome(string actionId, float outcome, ApiStatus apiStatus = null)
+        public bool TryReportOutcome(string actionId, object outcome, ApiStatus apiStatus = null)
         {
-            int result = LiveModelReportOutcome(this.NativeHandle, actionId, outcome, apiStatus.ToNativeHandleOrNullptr());
+            string objStr = JsonConvert.SerializeObject(outcome);
+            int result;
+            if(float.TryParse(objStr, out float temp))
+            {
+                result = LiveModelReportOutcome(this.NativeHandle, actionId, temp, apiStatus.ToNativeHandleOrNullptr());
+            }
+            else
+            {
+                result = LiveModelReportOutcome(this.NativeHandle, actionId, objStr, apiStatus.ToNativeHandleOrNullptr());
+            }
             return result == NativeMethods.SuccessStatus;
         }
+
+        public new void Dispose()
+        {
```

Comment:
What is the point of overloading this method if you don't have any new functionality besides "base.Dispose()"?

Reply:
the interface ILiveModel needs to have Dispose as it will be called from DecisionService. this Dispose goes and calls SafeHandle's Dispose method